app/models/group_order_article.rb

```diff
@@ -11,6 +11,8 @@ class GroupOrderArticle < ApplicationRecord
   validates_uniqueness_of :order_article_id, :scope => :group_order_id    # just once an article per group order
 
   scope :ordered, -> { includes(:group_order => :ordergroup).order('groups.name') }
+  default_scope { joins(:order_article => :article).order('articles.name') }
+
```

Comment:
I'm usually quite careful using default scopes, and joining wouldn't be in my book. Ran into cases where I had to used `unscoped` to make queries work.
Instead, why not adding a scope called e.g. `sorted` or `sorted_by_article_name` and use that in controllers?

Same applies to other default scopes.

Reply:
i think you're right - it had not caused me a problem until i used the pickups view - where the sorting conflicted with sorting by supplier and then by name.

it will take more changes now in the code though i think, to find all the usages, but i agree that this is the better way to do it.

rails is sometimes a little too convenient.  best to avoid those features that are quick fixes now but create a mess later.